wordpress-mobile/WordPress-Android
# Add Persistence Mechanism for App Logs

This PR makes progress towards issue #10698.

It addresses an issue found during review with a previous ([PR 10738](https://github.com/wordpress-mobile/WordPress-Android/pull/10738)).

### Description

This PR is not ready to merge yet, but I would like to get a bit of feedback on it to make sure that I am on the right track before continuing to work on it.

**I added the following classes to handle persistence using a `SQLite` database:**
- `LogDatabase`
- `LogTable`

The intent is to eventually move `LogDatabase` and `LogTable` to the Tracks library.  I temporarily added them to `org.wordpress.android.util` to be in the same place as the ` AppLog` class which uses it.

**I decided on using a SQLite database for two reasons:**
- It's a secure way to store data on Android.  This is important because log data can contain sensitive information.
- I can easily structure the data so that there is more flexibility in how it can be used.

Currently `AppLog` will load the log data for previous launches from the db when it first initializes. Then it will continue persisting log lines into the db as they are added.  `AppLog` will also load app info and device info for each previous launch session from the db and display those as a log entry that is not persisted.

### Testing Instructions

I will be looking into adding automated testing coverage as my next task. 🙂 
For now to test, the steps are:
1. launch the WordPress app on Android
2. goto `Me`->`Help & Support`->`Application log`
3. view log entries
4. close the WordPress app
5. launch the WordPress app
6. view log entries, notice the log entries from the previous launch is in the list.  After the last log entry from the previous launch, there should be an entry with the tag `APP_SESSION_END` that contains app and device info from the previous launch, as long as a session id for reference.

### Next Steps for the Persistence Mechanism 

- add automated testing coverage
- move the log persistence classes to Tracks
